### PR TITLE
#30: Creation of module for plotting scorecard metrics 

### PR DESCRIPTION
**Pull request by GabrielSGoncalves**
**Task**
Creation of module with functions to plot AUROC, Gini and KS for Scorecard models.

[#30](https://github.com/guillermo-navas-palencia/optbinning/issues/30)

**Checklist:**

- [x] Create module with functions.
- [x] Add functions to module \_\_init\_\_.
- [x] Check code style with Flake8.
- [x] Check branch modifications with pytest.

**Tasks that still need to be done**
The unit test for the `scorecard.plots.py` module were not implemented.

**Test Log**
```
======================================== test session starts ========================================
platform linux -- Python 3.7.7, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/gabriel/Documents/Repos/optbinning
collected 81 items                                                                                  

tests/test_binning.py ................                                                        [ 19%]
tests/test_binning_process.py .................                                               [ 40%]
tests/test_binning_scenarios.py ...........                                                   [ 54%]
tests/test_continuous_binning.py .........                                                    [ 65%]
tests/test_mdlp.py ....                                                                       [ 70%]
tests/test_multiclass_binning.py ........                                                     [ 80%]
tests/test_scorecard.py ................                                                      [100%]

========================================= warnings summary ==========================================
tests/test_binning_process.py::test_default_fit_transform_no_selected_variables
  /home/gabriel/miniconda3/envs/optbinning_scorecard/lib/python3.7/site-packages/optbinning-0.5.0-py3.7.egg/optbinning/binning/binning_process.py:759: UserWarning: No variables were selected: either the data is too noisy or the selection_criteria too strict.

tests/test_scorecard.py::test_estimator_not_coef
  /home/gabriel/miniconda3/envs/optbinning_scorecard/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
    return f(*args, **kwds)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================ 81 passed, 2 warnings in 123.68s (0:02:03) =============================
```

### CODE REVIEW

File: optbinning/scorecard/plots.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
+"""
+Modules with plots for visualizing model performance.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_curve, roc_auc_score
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:7]: Add space between import and from import.

File: optbinning/scorecard/plots.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
+"""
+Modules with plots for visualizing model performance.
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:2]: Module (remove s)

File: optbinning/scorecard/plots.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_curve, roc_auc_score
+
+
+def plot_auroc(
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:10]: - General comment: to be consistent with other classes, replace function arguments by:

```python

def plot_auc_roc(df, target_col, proba_col, title="ROC curve", xlabel="False positive rate",
                 ylabel="True positive rate", roc_line_color="blue", reference_line_color="black",
                 savefig=False, figname=None):

```
- Changes:
   - Function name: plot_auc_roc or plot_roc_auc. See: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score
   - Add line to color parameters.
   - figname is None by default, the user must provide a name.

File: optbinning/scorecard/plots.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+        Label for the x-axis.
+
+    ylabel : str (default='True positive rate')
+        Label for the y-axis.
+
+    roc_color : str (default='blue')
+        Color for the roc line.
+
+    reference_color : str (default='red')
+        Color for reference line.
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:48]: Color for the reference line.

File: optbinning/scorecard/plots.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    savefig : bool (default=False)
+        To save figure.
+
+    figname : str (default='auroc.fig')
+        Name for figure file.
+
+    Returns
+    -------
+    auroc : float
+        Calculate AUROC score.
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:59]: roc_auc_score: float
     Compute Area Under the Receiver Operating Characteristic Curve (ROC AUC) from prediction scores.

File: optbinning/scorecard/plots.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    target_col,
+    proba_col,
+    title='Gini',
+    xlabel='Cumulative % Population',
+    ylabel='Cumulative % Bad',
+    gini_color='blue',
+    reference_color='black',
+    savefig=False,
+    figname='gini.png',
+):
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:90]: See comments in plot_auroc

File: optbinning/scorecard/plots.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_curve, roc_auc_score
+
+
+def plot_auroc(
+    df,
+    target_col,
+    proba_col,
+    title='ROC curve',
```
Comment (guillermo-navas-palencia) [inline optbinning/scorecard/plots.py:14]: Use double quotes (``"``) for strings.

File: optbinning/scorecard/plots.py (hunk 1 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py:11]: Done.

File: optbinning/scorecard/plots.py (hunk 2 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py]: Should I add also your name to the module?

File: optbinning/scorecard/plots.py (hunk 3 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py]: Couldn't understand the "Add line to color parameters"

File: optbinning/scorecard/plots.py (hunk 4 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py]: ok.

File: optbinning/scorecard/plots.py (hunk 5 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py]: ok.

File: optbinning/scorecard/plots.py (hunk 7 above)
Comment (GabrielSGoncalves) [reply optbinning/scorecard/plots.py]: All fixed.

**Pull request merged**